backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/monitoringannotation/aspect/MonitorAspect.java

```diff
@@ -0,0 +1,135 @@
+
+import de.metas.monitoring.adapter.MicrometerPerformanceMonitoringService;
+import de.metas.monitoring.adapter.PerformanceMonitoringService;
+import de.metas.monitoringannotation.annotation.Monitor;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.element.api.AdWindowId;
+import org.adempiere.ad.window.api.IADWindowDAO;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import org.springframework.web.servlet.HandlerMapping;
+
+import javax.servlet.http.HttpServletRequest;
+import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+@Aspect
+@Component
+public class MonitorAspect
+{
+	private final MicrometerPerformanceMonitoringService service;
+
+	public MonitorAspect(@NonNull final MicrometerPerformanceMonitoringService service)
+	{
+		this.service = service;
+	}
+
+	@Value("${performance.monitoring.enable:false}")
+	private Boolean perfMonEnvVar;
+
+	@Around("execution(* *(..)) && @annotation(de.metas.monitoringannotation.annotation.Monitor)")
+	public Object monitorMethod(ProceedingJoinPoint pjp) throws Throwable
+	{
+		final PerformanceMonitoringService.Metadata metadata;
+		final Callable callable = getCallableFromProceedingJoinPoint( pjp );
+
+		if(perfMonEnvVar = false)
+		{
+			callable.call();
+		}
+
+		Method method = ((MethodSignature) pjp.getSignature()).getMethod();
+		Monitor monitorAnnotation = method.getAnnotation(Monitor.class);
+
+		if(monitorAnnotation.type() == PerformanceMonitoringService.Type.REST_CONTROLLER_WITH_WINDOW_ID)
+		{
+			HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest();
+			Map pathVariables = (Map) request.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+			String windowId = (String)pathVariables.get("windowId");
+			String window = (Services.get(IADWindowDAO.class).retrieveWindowName(AdWindowId.ofRepoId(Integer.parseInt(windowId)))).getDefaultValue();
+
+			metadata = PerformanceMonitoringService.Metadata.builder()
+					.name(pjp.getTarget().getClass().getSimpleName())
+					.type(monitorAnnotation.type())
+					.action(method.getName())
+					.window(window + " (" + windowId + ")")
+					.build();
+		}
+		else
+		{
+			metadata = PerformanceMonitoringService.Metadata.builder()
+					.name(pjp.getTarget().getClass().getSimpleName())
+					.type(monitorAnnotation.type())
+					.action(method.getName())
+					.build();
+		}
+
+		return service.monitor(callable, metadata);
+
+	}
+
+	private Callable<Object> getCallableFromProceedingJoinPoint(final ProceedingJoinPoint pjp)
```

Comment:
i guess this method is not needed.
anyways, if is needed, pls make it static and rename it to something like 'wrapAsCallable` because basically that's what it does